sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -20,6 +20,146 @@
 s3 = boto3.client("s3")
 
 
+def s3_event_handler(event, engine):
+    """Handler function for S3 events.
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    engine : sqlalchemy.engine.Engine
+        Engine object to interact with the database.
+
+    Returns
+    -------
+    dict
+        HTTP response
+    """
+    # Retrieve the Object name
+    s3_filepath = event["detail"]["object"]["key"]
+
+    filename = os.path.basename(s3_filepath)
+    # TODO: add checks for SPICE or other
+    # data types
+    science_file = ScienceFile(filename)
+
+    if science_file.is_valid is False:
```

Comment:
```suggestion
    if not science_file.is_valid:
```

Another idea would be to try and create your `ScienceFile` above and raise an exception and catch it here if it can't create it in the first place.

```python
class InvalidScienceFile(Exception):
    """Indicates a bad file type"""
    pass

try:
    science_file = ScienceFile(filename)
except InvalidScienceFile as e:
    return {"statusCode": 400, "body": str(e)}

Reply:
I like that idea. Will this mean that in path_helper.py, I will need to raise error instead of return bool, error_message when filename was not valid.

Reply:
I pushed changes for this change.